### PR TITLE
[mnet] Add new extractor

### PR DESCRIPTION
**Pull request by Kagami**

### CODE REVIEW

File: youtube_dl/extractor/mnet.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        file_url = rtmp_info['serverurl'] + rtmp_info['fileurl']
+
+        return {
+            'id': video_id,
+            'title': title,
+            'duration': duration,
+            'thumbnail': thumbnail,
+            'url': file_url,
+            # RTMP URL contains ".mp4" but resulting container is
+            # actually FLV.
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: rtmp's container is always flv by design. This comment is superfluous.

File: youtube_dl/extractor/mnet.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            'url': 'http://www.mnet.com/tv/vod/vod_view.asp?clip_id=172790&tabMenu=',
+            'md5': '1552891a585b9f640a64e4ba89d51f6e',
+            'info_dict': {
+                'id': '172790',
+                'ext': 'flv',
+                'title': '[직캠]일대일아이컨택ㅣ최유정 -  ♬Yum-Yum(얌얌) @ 콘셉트 평가',
+                'thumbnail': 're:^https?://.*\.jpg$',
+                'duration': 209,
+            },
+        },
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: Leave only one test if all test the same extraction scenario. For different URL schemas use `only_matching` test.

File: youtube_dl/extractor/mnet.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        info_url = 'http://content.api.mnet.com/player/vodConfig?id=%s' % video_id
+        info = self._download_json(info_url, video_id)
+        info = info['data']['info']
+
+        title = info['title']
+        duration = parse_duration(info.get('time'))
+        try:
+            thumbnail = info['cover']['maxres']['url']
+        except KeyError:
+            thumbnail = None
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: We usually use default fallbacks instead for shorter code: `info.get('cover', {}).get('maxres', {}).get('url')`

**Comment by Kagami** [comment]
@dstftw addressed all comments and squashed.

File: youtube_dl/extractor/mnet.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info_url = 'http://content.api.mnet.com/player/vodConfig?id=%s' % video_id
+        info = self._download_json(info_url, video_id)
+        info = info['data']['info']
+
+        title = info['title']
+        duration = parse_duration(info.get('time'))
+        thumbnail = info.get('cover', {}).get('maxres', {}).get('url')
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: There are multiple thumbnails available that can all be extracted.

**Comment by dstftw** [comment]
There is also an upload timestamp in `data/info/date` and age limit data in `data/info/adult`.

**Comment by Kagami** [comment]
@dstftw added few more fields from info JSON.

File: youtube_dl/extractor/mnet.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        description = info.get('ment')
+        duration = parse_duration(info.get('time'))
+        upload_date = unified_strdate(info.get('date'))
+        age_limit = 0 if info.get('adult', 'N') == 'N' else 18
+        thumbnails = [
+            {
+                'id': thumb_format,
+                'url': thumb.get('url'),
+                'width': int(thumb.get('width', 0)),
+                'height': int(thumb.get('height', 0)),
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: `utils.int_or_none` should be used instead.

File: youtube_dl/extractor/mnet.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        video_id = self._match_id(url)
+        info_url = 'http://content.api.mnet.com/player/vodConfig?id=%s' % video_id
+        info = self._download_json(info_url, video_id)
+        info = info['data']['info']
+
+        title = info['title']
+        description = info.get('ment')
+        duration = parse_duration(info.get('time'))
+        upload_date = unified_strdate(info.get('date'))
+        age_limit = 0 if info.get('adult', 'N') == 'N' else 18
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: `age_limit` should be `None` when `adult` is missing.

File: youtube_dl/extractor/mnet.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info_url = 'http://content.api.mnet.com/player/vodConfig?id=%s' % video_id
+        info = self._download_json(info_url, video_id)
+        info = info['data']['info']
+
+        title = info['title']
+        description = info.get('ment')
+        duration = parse_duration(info.get('time'))
+        upload_date = unified_strdate(info.get('date'))
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: `utils.parse_iso8601` should be used instead and extracted as `timestamp`.

File: youtube_dl/extractor/mnet.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                'url': thumb.get('url'),
+                'width': int(thumb.get('width', 0)),
+                'height': int(thumb.get('height', 0)),
+            }
+            for (thumb_format, thumb) in info.get('cover', {}).items()
+        ]
+
+        rtmp_info_url = info['cdn'] + 'CLIP'
+        rtmp_info = self._download_json(rtmp_info_url, video_id)
+        file_url = rtmp_info['serverurl'] + rtmp_info['fileurl']
```
Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: Video URL should be extracted the very first right after the title.

Comment (Kagami) [inline youtube_dl/extractor/mnet.py]: Explain please. What do you want to see?

Comment (dstftw) [inline youtube_dl/extractor/mnet.py]: Move in right after `title = info['title']`.

**Comment by Kagami** [comment]
Addressed all comments.

**Comment by dstftw** [comment]
Cherry-picked in e0317686666f9de4a6eca3fc26ede32e664f2bec, thanks.
Do you want to be credited as one of the authors of youtube-dl? If so under what name?

**Comment by Kagami** [comment]
Thanks. "Kagami Hiiragi" will be fine.

**Pull request closed without merging**